### PR TITLE
Added ZIP support spec

### PR DESCRIPTION
**Pull request by midoriiro**
Added ZIP support spec for community review as draft

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/1717)

### CODE REVIEW

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
Looks like this PR's touching files (467,403 additions and 467,269 deletions across 1314 different files) that shouldn't which makes reviewing changes a bit difficult.

**Comment by midoriiro** [comment]
> 
> 
> Looks like this PR's modifying files (467,403 additions and 467,269 deletions across 1314 different files) which makes reviewing changes a bit difficult.

Exact, did not notice that. I just clone with git bash. I'll try to fix that.

File: doc/specs/#140 - ZIP Support.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+For [#140](https://github.com/microsoft/winget-cli/issues/140)
+
+## Abstract
+
+This feature adds to winget-cli the ability to install application delivered through a ZIP archive.
+
+## Inspiration
+
+A bunch of applications are only provided by developer's through a ZIP archive. There are three cases:
+1) The archive contain only one binary that need do to be extracted and add to PATH env var
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:19]:
```suggestion
1) The archive contains only one binary (a portable / standalone executable).
```

File: doc/specs/#140 - ZIP Support.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+2) The archive contain an installer that need to be executed and let do his job
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:20]:
```suggestion
2) The archive contains only one supported installer (msix, msi, or exe).
```

File: doc/specs/#140 - ZIP Support.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+3) The archive contain a complex hierarchy that need do to be extracted and add specific folder to PATH env var
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:21]:
```suggestion
3) The archive contains more than one file,
```

File: doc/specs/#140 - ZIP Support.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+## Solution Design
+
+A new installation flow need to be created to execute different installation strategies regarding these three cases above.
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:25]:
```suggestion
A new installation flow needs to be created to execute different installation strategies regarding these three cases above.
```

File: doc/specs/#140 - ZIP Support.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+First thing first, the archive need to be extracted in a specific path (TEMP folder can be a good candidate).
+
+Regarding case 1, installation flow check if TEMP folder contain a single binary and repack it
+into an MSI package (including setting PATH env var). Then, MSI installation flow can take control.
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:30]: Packaging a standalone / portable application is an interesting approach. I'm not sure we had considered this specific approach. We may prefer the concept of repackaging as an MSIX as that is the latest technology, but there may be different concerns to discuss here.

File: doc/specs/#140 - ZIP Support.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+This case is the complex one, example with 'vagrant' hierarchy:
+- root
+  - bin 
+    - vagrant.exe
+    - other.exe
+  - embedded
+    - bin
+    - some other folders
+
+This hierarchy can be packed into an MSI installer, but how can we specify which binary to add
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:45]: This use case is why packaging an MSIX may be better than an MSI, but we will need to dig a bit deeper here.

File: doc/specs/#140 - ZIP Support.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    - vagrant.exe
+    - other.exe
+  - embedded
+    - bin
+    - some other folders
+
+This hierarchy can be packed into an MSI installer, but how can we specify which binary to add 
+in PATH env var (in that case bin/vagrant.exe)?
+
+Take a look into the following manifest examples (v1.1.0)
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:48]: We should stick with only supporting this in the 1.1.0 or later schema.

File: doc/specs/#140 - ZIP Support.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+Installers:
+- Architecture: x64
+  InstallerUrl: https://editor.com/app/1.0.zip
+  InstallerSha256: ...
+  BinaryPath: bin/binary.exe
+ManifestType: installer
+ManifestVersion: 1.0.0
+```
+
+This is not compliant with the actual schema v1.1.0. We need to specify which binary to add into
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:95]: If we need additional metadata, this would be a newer schema than 1.1.0. Once we sufficiently capture all the use cases, we can certainly look at extending the schema. The clients lacking support for the 1.1.0 or later schema must be able to determine this is not going to be a valid scenario, so the .zip would not be a supported installer type on those clients.

File: doc/specs/#140 - ZIP Support.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+PATH env var. Therefore, schema need to be updated to take knowledge of that information.
+
+### Repack ZIP into MSI package
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:98]:
```suggestion
### Repack uncompressed files into a supported package format
```

File: doc/specs/#140 - ZIP Support.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+This section need more specifications. There are many tool to repack an MSI package, some expansive and
+some apparently free (Advanced Installer ?).
+
+## UI/UX Design
+
+One downside of this feature is the MSI repackaging will be executed on client side.
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:105]:
```suggestion
One downside of this feature is the package will be built on the client side.
```

File: doc/specs/#140 - ZIP Support.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+Which mean for each installation the same process will be executed and remain the same for an
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:106]: This will certainly have performance implications. We also need to ensure rational versioning so the upgrade flow also works as expected.

File: doc/specs/#140 - ZIP Support.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+This section need more specifications. There are many tool to repack an MSI package, some expansive and
+some apparently free (Advanced Installer ?).
+
+## UI/UX Design
+
+One downside of this feature is the MSI repackaging will be executed on client side.
+Which mean for each installation the same process will be executed and remain the same for an 
+application x.y.z.
+
+But with a proper MSI repacking solution, this operation should be transparent to the end-user.
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:109]: We would likely have some output for the user with the packaging step so they at least get an indication of progress.

File: doc/specs/#140 - ZIP Support.md [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+
+## Capabilities
+
+### Accessibility
+
+This should have no direct impact on accessibility.
+
+### Security
+
+No direct impact on security.
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:119]: There would most certainly be security implications. We may even need a SHA256 hash for each of the files to be packaged. We don't want to be in a state where the files are downloaded and modified prior to the packaging steps. This will also have a substantial impact on the validation flow so we can verify the hashes are valid at validation time, and correctly included in the manifest for any changes to the manifest.

File: doc/specs/#140 - ZIP Support.md [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+
+### Reliability
+
+This is not expected to impact reliability.
+
+### Compatibility
+
+No code breaking, reuse existing code and add code to support zip installation.
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:127]: This would not be a backward compatible behavior. Only clients with the support for this feature would be able to install packages of this type. We would need to add something like "zip" to the enumeration of installer types. The initial implementation would also be an experimental feature until we have covered enough of the different use cases to confirm viability of the solution.

File: doc/specs/#140 - ZIP Support.md [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+
+### Performance, Power, and Efficiency
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:129]: Building the installer is likely to take additional time and resources on the client. There will be performance considerations here with the client in the packaging phase.

File: doc/specs/#140 - ZIP Support.md [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+
+## Potential Issues
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:131]: MSIX packages are typically signed. There would likely be security issues related to signing. If the MSI route or some other mechanism is used for the package, we would also need to look at those considerations. Even if we just move the standalone / portable application (or set of files) to a directory path, any changes to those files could have unintended consequences. Some packages also generate additional files in their directories so those may need to be considered during an upgrade or uninstall flow.

File: doc/specs/#140 - ZIP Support.md [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+
+## Future considerations
```
Comment (denelon) [inline doc/specs/#140 - ZIP Support.md:133]: Other compression formats may also need to be considered in the future. The initial implementation is likely to be limited to .zip files.

**Comment by denelon** [comment]
Thanks for taking the time to write this draft PR! Please be patient as we're dealing with limited resources during this time of year. We also need time for additional internal design review, and have a security review performed to make sure we aren't facing any other additional security concerns. There may also be licensing or copyright concerns to consider if the publisher or the artifacts aren't governed by an open source license.

File: doc/specs/#140 - ZIP Support.md (hunk 5 above)
Comment (JohnMcPMS) [reply doc/specs/#140 - ZIP Support.md:30]: I'm not sure that writing code to transform the archive into an installable type is better than simply writing the code to manage archives directly.  All of the proposals require unpacking the archive, which is itself the primary step in management.  Afterward, adding to the `%PATH%` is fairly straightforward and the issues are more about lifetime management.  Injecting our own ARP entry is largely sufficient to achieve that.

File: doc/specs/#140 - ZIP Support.md (hunk 8 above)
Comment (JohnMcPMS) [reply doc/specs/#140 - ZIP Support.md:95]: Currently you would get an `ERROR_NOT_SUPPORTED` result.  If you think that isn't a good experience, we should come up with a better one and implement now to increase the chances of the downlevel client having it.

File: doc/specs/#140 - ZIP Support.md [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+InstallModes: ...
+InstallerSwitches:
+  Silent: ...
+  SilentWithProgress: ...
+UpgradeBehavior: install
+Installers:
+- Architecture: x64
+  InstallerUrl: https://editor.com/app/1.0.zip
+  InstallerSha256: ...
+  BinaryPath: bin/binary.exe
```
Comment (JohnMcPMS) [inline doc/specs/#140 - ZIP Support.md:90]: If this is intended to be the `%PATH%` update location, it should be to a directory and not a file.  The default would be either be no `%PATH%` update or the root.

File: doc/specs/#140 - ZIP Support.md [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+  - embedded
+    - bin
+    - some other folders
+
+This hierarchy can be packed into an MSI installer, but how can we specify which binary to add 
+in PATH env var (in that case bin/vagrant.exe)?
+
+Take a look into the following manifest examples (v1.1.0)
+
+### Case 1|2 YAML manifest
```
Comment (JohnMcPMS) [inline doc/specs/#140 - ZIP Support.md:50]: I would argue for case 2 to be the one that includes a relative path to the installer, in the event that there are support files included within the archive.

File: doc/specs/#140 - ZIP Support.md (hunk 3 above)
Comment (JohnMcPMS) [reply doc/specs/#140 - ZIP Support.md:21]: The cases I see:
1. Archive is the set of files that should be placed on the system (both your 1 and 3)
2. Archive is a container for the installer, and is itself temporary (your 2)
3. [New] Archive is basically both of the above; the files themselves need to be kept, but there is some registration step that is needed.

File: doc/specs/#140 - ZIP Support.md (hunk 6 above)
Comment (jedieaston) [reply doc/specs/#140 - ZIP Support.md:45]: MSIX packaging client side is impossible right now due to the signing requirements, no? Right now, you either have to sign in with Azure AD to get a short lived (48 hours?) cert, or buy a longer one.

File: doc/specs/#140 - ZIP Support.md (hunk 16 above)
Comment (jedieaston) [reply doc/specs/#140 - ZIP Support.md:131]: > if we just move the standalone / portable application (or set of files) to a directory path, any changes to those files could have unintended consequences. 

which is true of any package winget currently installs, as with admin rights (or for user scoped packages, normal rights) I can go mess around with anything the installer created. I think that might be a separate issue.

File: doc/specs/#140 - ZIP Support.md (hunk 5 above)
Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:30]: @denelon I'm not an expert concerning Windows packaging application. The main idea of this specification is to package the content of a zip file (except for case 2) in order to let winget managing the lifecycle of the new package. 

I saw Windows SDK offers a set of CLI command to create MSIX package, but that mean installing Windows SDK.

@JohnMcPMS  I don't know what is ARP, but if I understood your suggestion. How you do you handle uninstall/upgrade of the package ? That the goal of an ARP entry ?

File: doc/specs/#140 - ZIP Support.md (hunk 13 above)
Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:119]: That could be tricky to add all checksum for all files in the manifest. If the ZIP archive is validated can we just lock the extracted files before packaging to prevent any modifications by other process ?

File: doc/specs/#140 - ZIP Support.md (hunk 14 above)
Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:127]: Yes, you are right. For the moment this draft is only intended to schema 1.1.0. Like i suggested, we maybe need to update the schema (1.1.1?) to handle case 3 and other new cases.

File: doc/specs/#140 - ZIP Support.md (hunk 16 above)
Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:131]: Like someone said on issue #140, if a file is modified is not the packager responsibility but it's an user fault. If files are created by a process delivered by the package, these files should be referenced somewhere to prevent overwriting them, or rather we can specify to overwrite theses files during an upgrade. Typically for config files, user need to edit them, that could be a problem if on each upgrade these files are overwritten. 

Concerning signing a MSIX package, A self-signed package could be acceptable ?

File: doc/specs/#140 - ZIP Support.md (hunk 18 above)
Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:90]: Yes, I noticed my mistake before your write your comment ^^".  You are right, BinaryPath should be a relative path from the root archive.

File: doc/specs/#140 - ZIP Support.md (hunk 19 above)
Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:50]: Case 2 expect to have an installer at the root archive. But if the installer is located to 'archive.zip/somefolder/installer.exe' that should be a new case in my opinion. To avoid having more case I suggest to add a new attribute like BinaryPath.

**Comment by midoriiro** [comment]
@denelon I know this feature is not planned to be released shortly. However, I'll take on consideration your feedback (@JohnMcPMS included) and push a new revision soon, so we can discuss it when you will be available ;-)

File: doc/specs/#140 - ZIP Support.md (hunk 11 above)
Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:106]: You mean that information (like version, name, etc) from the winget manifest should be linked to built MSI(X) package ?

File: doc/specs/#140 - ZIP Support.md (hunk 5 above)
Comment (denelon) [reply doc/specs/#140 - ZIP Support.md]: Add Remove Programs (ARP) is where many of the entries for programs installed on Windows 10 and Windows 11 get recorded. We use that metadata for the list command and as the source for values we used to correlate with manifests in configured sources. We don't necessarily need to create an "insaller" per se'. It's possible to just write entries to ARP (now part of Apps & Features). The Windows Package Manager could act as an "installer" proxy to create the entries and modify the path environment variable. We're looking at pros and cons to several approaches.

File: doc/specs/#140 - ZIP Support.md [hunk 20]
```diff
@@ -0,0 +1,180 @@
...
+InstallModes: ...
+InstallerSwitches:
+  Silent: ...
+  SilentWithProgress: ...
+UpgradeBehavior: install
+Installers:
+- Architecture: x64
+  InstallerUrl: https://editor.com/app/1.0.zip
+  InstallerSha256: ...
+  BinaryPath: bin (e.g. bin folder where vagrant.exe is located in my example)
```
Comment (Trenly) [inline doc/specs/#140 - ZIP Support.md:120]: Could this be simplified to "InstallerPath" as well, and have the cli determine if it is an installer or binary?

Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:120]: Case 2 cover installer (.exe, .msi, etc). You're pointing case 3, which is for complex hierarchy that does not include installer, only binary with complex folders structure. The base idea is to implement all the cases. Nonetheless, "BInaryPath" maybe not a good name, any idea ?

File: doc/specs/#140 - ZIP Support.md (hunk 5 above)
Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md]: I see, it's make sense. I looked if an API exist about managing an ARP entry, but I did found nothing. If winget-cli already manage that kind of tasks, @JohnMcPMS made a good point.

I don't know what is the state of art about ARP in Windows 10/11 now. But, ten years ago I wrote an application that find all installed applications on the system through the registry, the main goal was to uninstall silently a batch of applications. That was tricky to handle all the cases. 

If an API already exist, the repacking solution would not be the best approach(in my opinion) as you explained both of you.

I'll try to find some documentation on the internet and this repository to study how to manage an ARP entry.

File: doc/specs/#140 - ZIP Support.md (hunk 20 above)
Comment (Trenly) [reply doc/specs/#140 - ZIP Support.md:120]: I’m saying that for case 2, you are suggesting that a key be added to define the path to the exe/msi/etc.

Why can’t that same key be used for case 3?

File: doc/specs/#140 - ZIP Support.md [hunk 21]
```diff
@@ -0,0 +1,180 @@
...
+InstallModes: ...
+InstallerSwitches:
+  Silent: ...
+  SilentWithProgress: ...
+UpgradeBehavior: install
+Installers:
+- Architecture: x64
+  InstallerUrl: https://editor.com/app/1.0.zip
+  InstallerSha256: ...
+  InstallerPath: somefolder/installer.exe
```
Comment (Trenly) [inline doc/specs/#140 - ZIP Support.md:95]: I think we may also need a key defining the secondary installer type.

It is likely that some of the exe’s shipped inside zip packages are actually nullsoft or inno, which should be preferred to exe since the CLI already handles their silent switches without having to define them

File: doc/specs/#140 - ZIP Support.md (hunk 20 above)
Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:120]: I got confused you was quoting case 3. On case 3, BinaryPath key is for indicate to register a path after the zip extraction into user or system environment variables (need to be defined how to). 

Correct me if I'm wrong, but from my point of view regarding case 2 (excluding my example manifest 2.1). Case 2 assuming that the zip archive only contains one file, the installer. Case 2 also assuming that the installer will set the proper environment variables if needed (specifically $PATH) or not. that depend of the nature of the application and how editors package their applications. I don't know, how do you manage that kind of thing, winget-cli is agnostic regarding this aspect ?

To answer your question, case 3 does not handle installer file. That the job of case 2 and 2.1. Case 2 does not need to define InstallerPath key because, the archive only contains the installer at the root archive. On the other hand, case 2.1 zip archive maybe not contains only one file and maybe not located at root, so InstallerPath key is needed to indicate where to find the installer after the extraction and execute it.

To summarize, BinaryPath and InstallerPath are optional keys that depend on the context and they should be set properly by the package maintainer. To avoid messing up with these settings, updating the documentation and the helper script YAMLCreate.ps1 should be needed, in my opinion. For all cases, if the manifest is miss-configured the procedure should fail in an organic way. 

I don't know if I'm clear, my english is not perfect. Maybe I should clarify these aspects on the specification.

File: doc/specs/#140 - ZIP Support.md (hunk 21 above)
Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:95]: A key definition already exist on schema [1.1.0](https://github.com/midoriiro/winget-cli/blob/03f8a555693003129e0199759135d874a580a0b5/schemas/JSON/manifests/v1.1.0/manifest.installer.1.1.0.json#L459) (line 459)

Maybe we can use it to tell which installation flow to choose?

Comment (Trenly) [reply doc/specs/#140 - ZIP Support.md:95]: That key, by definition, would be "zip" since the primary installer is inside of a .zip file. The issue I'm trying to get ahead of is that the zip file could contain different types of exes which have their own installation flow

Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:95]: The key I'm pointing could be resolve that issue with something like this:

```
PackageIdentifier: Editor.App
PackageVersion: 1.0
MinimumOSVersion: 10.0.0.0
InstallerType: zip
Scope: ...
InstallModes: ...
InstallerSwitches:
  Silent: ...
  SilentWithProgress: ...
UpgradeBehavior: install
Installers:
- Architecture: x64
  InstallerUrl: https://editor.com/app/1.0.zip
  InstallerSha256: ...
  InstallerPath: somefolder/installer.exe
  InstallerType: inno # or something else
ManifestType: installer
ManifestVersion: 1.0.0
```

Comment (Trenly) [reply doc/specs/#140 - ZIP Support.md:95]: I don't believe that is valid. Having the key at the manifest level is not compatible with the key at the installer level. One of them will override the other.

Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:95]: I see, which component validate the manifest and could be responsible of overriding these key ? An other component or the installation flow ?

Comment (Trenly) [reply doc/specs/#140 - ZIP Support.md:95]: I'm not sure I understand your question.

The overriding is an inherent property of the CLI. If a key is specified at the installer level, it is overridden by the manifest level. You can't have them defined at both levels
```
PackageIdentifier: ...
PackageVersion: ...
InstallerType: zip <----- THIS IS THE MANIFEST LEVEL
Installers:
- Architecture: ...
  InstallerUrl: ...
  InstallerSha256: ...
  InstallerType: nullsoft<----- THIS IS THE INSTALLER LEVEL
ManifestType: installer
ManifestVersion: 1.0.0
```

Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:95]: Ok i see, that's a dead end. About the issue, can you provide a concrete example with multiple installers included in a zip archive ? That could help to better understanding and had a better overview on this issue.

Comment (Trenly) [reply doc/specs/#140 - ZIP Support.md:95]: I’m not saying multiple installers may be in a .Zip archive. I’m saying a manifest may have different types of installers, and a zip archive may be one of them.

For example - the 7-zip manifest has installers for both exe and msi. Both are in the same manifest.

Some publishers pubish exe’s, msi’s, and zip files, which should all be able to be in a single manifest

Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:95]: Ok I understand now. And I agree with you about defining another key to specify the installer type. Maybe InstallerSubType ?

I need to clarify some points on the specification about previous conversations. I'll commit soon.

Comment (denelon) [reply doc/specs/#140 - ZIP Support.md:95]: https://github.com/microsoft/winget-cli/issues/1242 is how we're planning to capture some of the "nested installer type" issues. For example, PowerToys is distributed as an .exe, but it installs as an .msi. The installer is a wrapper to help with dependencies.

Comment (Trenly) [reply doc/specs/#140 - ZIP Support.md:95]: Perfect, I think that as long as that issue is referenced in the spec, then there won't be an issue here

Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md:95]: That should close this issue. Thank for your feedback.

File: doc/specs/#140 - ZIP Support.md (hunk 5 above)
Comment (JohnMcPMS) [reply doc/specs/#140 - ZIP Support.md]: I don't think that there are APIs; we certainly do not use them and I looked for them when we implemented reading.  It is basically a registry data contract.

Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md]: Yes, I saw that yesterday. I updated the spec. There some points to discuss about it but, especially where to install the zip content. The ARP section seem good for you ?

Comment (parithon) [reply doc/specs/#140 - ZIP Support.md]: Perhaps we treat ZIP installs the same way visual studio code is installed (i.e. user installer) within the AppData folder.

Comment (midoriiro) [reply doc/specs/#140 - ZIP Support.md]: That I suggested in the last revision of the spec. For user scope, application will be installed in AppData/Local/myapplication. But maybe there is a better location for user scope ? 

In any ways, scope and location can be overrides by users when calling winget install. Just found that, that should be take in consideration on the specification.

**Comment by midoriiro** [comment]
What's the state about my last commits? Someone has time to review them? I know you're busy, I'm just checking. Thanks

**Comment by denelon** [comment]
We're getting ramped up on how we're going to tackle .zip support. We will take as much of what has been drafted as possible.

**Comment by ryfu-msft** [comment]
Hey @midoriiro,

The team has begun working on a design for adding .ZIP installer support. As this spec is a great starting point and a lot of important discussions have taken place, we plan to discuss and incorporate key components that you have brought to our attention into our design review. We will continue to update the community as we make progress. Great work :)

**Comment by denelon** [comment]
We're getting closer with support for portable applications:

* #182 

Once that is "code complete", we will turn our attention back to .zip.
I just wanted to provide an update 😊

**Comment by JohnMcPMS** [comment]
Superseded by #2270
